src/objects/core/zcl_abapgit_objects_files.clas.abap

```diff
@@ -248,6 +243,26 @@ CLASS zcl_abapgit_objects_files IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD is_json_metadata.
+
+    DATA lv_regex TYPE string.
+
+    FIELD-SYMBOLS <ls_file> LIKE LINE OF mt_files.
+
+    lv_regex = '^[^\.]+\.' && to_lower( ms_item-obj_type ) && '\.json$'.
+
+    LOOP AT mt_files ASSIGNING <ls_file>.
+      FIND REGEX lv_regex IN  <ls_file>-filename.
```

Comment:
the regex seems overkill? its just `CP |*.{ to_lower( ms_item-obj_type ) }.json|` ?